bazel/envoy_build_system.bzl

```diff
@@ -21,6 +21,7 @@ load(
 )
 load(
     ":envoy_test.bzl",
+    _envoy_cc_benchmark_binary_and_test = "envoy_cc_benchmark_binary_and_test",
```

Comment:
Nit: prefer `envoy_cc_benchmark`, this is a bit cleaner visually and implies the same information.

Reply:
I think I'll split this into 2 build extensions: envoy_cc_benchmark and envoy_benchmark_test.  It does require people adding both to BUILD files, but feels cleaner.

Reply:
Up to you, I'm fine with just renaming to `envoy_cc_benchmark` and having multiple targets.

Reply:
Switched to separate benchmark and test rules.